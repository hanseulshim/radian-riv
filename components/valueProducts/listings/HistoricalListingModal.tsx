import React, { useEffect, useState } from 'react'
import Modal from 'components/common/Modal'
import HistoricalListingTable from './HistoricalListingTable'
import HistoricalListingZoomPhotoModal from './HistoricalListingZoomPhotoModal'
import GalleryModal from './GalleryModal'
import ListingHistoryTable from './ListingHistoryTable'
import {
  getHistoricalListingHistory,
  getHistoricalListingProperty,
  HistoricalListingPropertyInterface,
  HistoricalListingInterface
} from 'api'
import { useOrder } from 'context/OrderProvider'

interface Props {
  closeModal: () => void
  propertyId: string
  title: string
}

export default function HistoricalListingModal({
  closeModal,
  propertyId,
  title
}: Props) {
  const { order } = useOrder()
  const [galleryModal, setGalleryModal] = useState(false)
  const [zoomPhoto, setZoomPhoto] = useState(false)
  const [
    currentProperty,
    setCurrentProperty
  ] = useState<HistoricalListingPropertyInterface>({
    address: null,
    bath: null,
    bed: null,
    garage: null,
    listDate: null,
    lotSize: null,
    mlsComments: null,
    mlsName: null,
    saleType: null,
    sqft: null,
    year: null,
    zip: null,
    photos: []
  })
  const [listingHistory, setListingHistory] = useState<
    HistoricalListingInterface[]
  >([])

  useEffect(() => {
    const getCurrentProperty = async () => {
      const property = await getHistoricalListingProperty(propertyId, order.id)
      setCurrentProperty(property)
    }
    getCurrentProperty()
    const listingHistory = async () => {
      const history = await getHistoricalListingHistory(propertyId, order.id)
      setListingHistory(history)
    }
    listingHistory()
  }, [])

  return (
    <Modal
      closeModal={closeModal}
      title={''}
      percent={95}
      id="historical-listing-modal"
    >
      <div className="top-row">
        <h2>{title}</h2>
        <button onClick={() => window.print()} className="btn btn-icon dark">
          <span>Print</span>
          <span className="icon-container">
            <img src={`${process.env.baseUrl}/images/print.svg`} alt="print" />
          </span>
        </button>
      </div>
      <div className="listing-info">
        <div className="listing-image">
          <div className="image-container">
            {currentProperty && currentProperty.photos[0] ? (
              <img
                className="image"
                src={currentProperty.photos[0]}
                alt="property"
              />
            ) : (
              <img
                className="image"
                src={`${process.env.baseUrl}/images/photo-unavailable.svg`}
                alt="unavilable"
              />
            )}
            <div className="overlay" onClick={() => setZoomPhoto(true)}>
              <img
                src={`${process.env.baseUrl}/images/search.svg`}
                alt="print"
              />
            </div>
          </div>
          <button
            onClick={() => setGalleryModal(true)}
            className="btn btn-small"
          >
            View All Photos
          </button>
        </div>
        <div>
          {currentProperty && (
            <HistoricalListingTable tableData={[currentProperty]} />
          )}
          <div className="info-container">
            <div style={{ flex: 0.5, marginRight: '1em' }}>
              <div className="label">Address:</div>
              <div>{currentProperty?.address || null}</div>
              <div>{currentProperty?.zip || null}</div>
              <div className="label space">Listing Date:</div>
              <div>{currentProperty?.listDate || null}</div>
              <div className="label space">MLS Name:</div>
              <div>{currentProperty?.mlsName || null}</div>
            </div>
            <div style={{ flex: 1 }}>
              <div className="label">MLS Comments:</div>
              <div>{currentProperty?.mlsComments || null}</div>
            </div>
          </div>
        </div>
      </div>
      <div className="listing-history">
        <h2>Listing History</h2>
        {listingHistory && (
          <ListingHistoryTable
            tableData={listingHistory}
            setListingHistory={setListingHistory}
          />
        )}
      </div>
      {galleryModal && (
        <GalleryModal
          photos={currentProperty.photos}
          closeModal={() => {
            setGalleryModal(false)
          }}
        />
      )}
      {zoomPhoto && (
        <HistoricalListingZoomPhotoModal
          photo={currentProperty.photos[0]}
          closeModal={() => {
            setZoomPhoto(false)
          }}
        />
      )}
    </Modal>
  )
}
