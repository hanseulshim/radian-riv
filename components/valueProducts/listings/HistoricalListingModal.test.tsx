import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import HistoricalListingModal from './HistoricalListingModal'
describe('HistoricalListingModal', () => {
  let component
  beforeAll(() => {
    const props = {
      closeModal: () => {},
      resultsId: '123',
      title: 'SDF'
    }
    component = shallow(<HistoricalListingModal {...props} />)
  })
  it('contains compnent classes', () => {
    expect(component.find('.top-row')).toHaveLength(1)
    expect(component.find('.listing-info')).toHaveLength(1)
    expect(component.find('.listing-image')).toHaveLength(1)
    expect(component.find('.info-container')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
