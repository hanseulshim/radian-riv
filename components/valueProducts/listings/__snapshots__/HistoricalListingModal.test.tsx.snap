// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoricalListingModal should match snapshot 1`] = `
<Modal
  closeModal={[Function]}
  id="historical-listing-modal"
  percent={95}
  title=""
>
  <div
    className="top-row"
  >
    <h2>
      SDF
    </h2>
    <button
      className="btn btn-icon dark"
      onClick={[Function]}
    >
      <span>
        Print
      </span>
      <span
        className="icon-container"
      >
        <img
          alt="print"
          src="undefined/images/print.svg"
        />
      </span>
    </button>
  </div>
  <div
    className="listing-info"
  >
    <div
      className="listing-image"
    >
      <div
        className="image-container"
      >
        <img
          alt="unavilable"
          className="image"
          src="undefined/images/photo-unavailable.svg"
        />
        <div
          className="overlay"
          onClick={[Function]}
        >
          <img
            alt="print"
            src="undefined/images/search.svg"
          />
        </div>
      </div>
      <button
        className="btn btn-small"
        onClick={[Function]}
      >
        View All Photos
      </button>
    </div>
    <div>
      <HistoricalListingTable
        tableData={
          Array [
            Object {
              "address": null,
              "bath": null,
              "bed": null,
              "garage": null,
              "listDate": null,
              "lotSize": null,
              "mlsComments": null,
              "mlsName": null,
              "photos": Array [],
              "saleType": null,
              "sqft": null,
              "year": null,
              "zip": null,
            },
          ]
        }
      />
      <div
        className="info-container"
      >
        <div
          style={
            Object {
              "flex": 0.5,
              "marginRight": "1em",
            }
          }
        >
          <div
            className="label"
          >
            Address:
          </div>
          <div />
          <div />
          <div
            className="label space"
          >
            Listing Date:
          </div>
          <div />
          <div
            className="label space"
          >
            MLS Name:
          </div>
          <div />
        </div>
        <div
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <div
            className="label"
          >
            MLS Comments:
          </div>
          <div />
        </div>
      </div>
    </div>
  </div>
  <div
    className="listing-history"
  >
    <h2>
      Listing History
    </h2>
    <ListingHistoryTable
      setListingHistory={[Function]}
      tableData={Array []}
    />
  </div>
</Modal>
`;
